x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -105,9 +122,7 @@ functionExpression
     ;
 
 functionName
-    // Additional function identifiers that are already a reserved word in the language
-    : MATCH
```

Comment:
Removing MATCH as a reserved word means we can simplify this. Leaving the `functionName` rule just in case we want to add a function named as a token.